resolwe_bio/processes/differential_expression/deseq.yml

```diff
@@ -60,9 +60,14 @@
         re-error "Read counts are required when using DESeq2."
       {% endif %}
 
-      {% if expressions|selectattr('source', 'equalto', (case|first).source)|list|length != expressions|length %}
-        re-error "DE analysis must be crated using the expression data from the same source."
-      {% endif %}
+      {% for c in control %}
+        {% for t in case %}
+          {% if control.0.source != c.source or c.source != t.source %}
+            re-warning "Genes in all expression data must be annotated by the same genome database."
+            re-error "Sample {{ c|name }} has {{ c.source }} gene IDs, while {{ t|name }} has {{ t.source }}."
```

Comment:
Missing error message for the first part of the condition in the `if` statement.